apso/src/main/scala/eu/shiftforward/apso/io/FileDescriptor.scala

```diff
@@ -52,6 +52,16 @@ trait FileDescriptor {
   def upload(localTarget: LocalFileDescriptor): Boolean
 
   /**
+   * Uploads an input stream to this file's location.
+   * The target must point to a file.
+   *
+   * @param inputStream the input stream that should be uploaded.
+   * @param length the input stream length (leaving this as `None` can have performance implications)
+   * @return `true` if the upload was successful, `false` otherwise.
+   */
+  def upload(inputStream: InputStream, length: Option[Long]): Boolean
```

Comment:
What do you think of adding a default value of `None` to length? I understand the performance implications of having it as `None` in some implementations, but forcing the call to `upload` to have a `length` argument can be cumbersome for implementations that don't really need it.